src/adapters/ingresses/apig.rs

```
@@ -51,6 +106,15 @@ async fn read_file(artifact_path: PathBuf) -> Result<Vec<u8>> {
 #[async_trait]
 impl Ingress for AwsApiGateway {
     async fn deploy(&mut self) -> Result<()> {
-        todo!()
+        // First, we need to deploy the new version of the lambda
+        let lambda_version = self.upload_lambda("releases").await.unwrap();
```

OH! You mentioned just using `unwrap()` everywhere to solve you problems. Not so! I think you want to replace `unwrap` with `?` in most places. Certainly here!

Unwrap just crashes the program if there's an error. Instead, we want to return the error to the user. That's why this function takes a `Result` -- because it could succeeds with `Ok` or it could return an error.

The question mark operator functions as an early return. If your function returns `Result<T, E>`, and you suffix `?` after the value, the Rust will try to convert your value into `E`. It does this by calling `.into()` on your value. So as long as your value implements `Into<E>`, it will Just Work™️.

What complicates this: we're not using `std::result::Result`. We're using a fancy, colorful version called `miette::Result`, which is slightly different. Lots of errors in the std lib aren't directly compatible with `miette::Result`, but you can almost always upgrade them with `.into_diagnostic()`.

TL;DR: If you find yourself reaching for `unwrap`, first try `?`. If that doesn't work, try `.into_diagnostic()?`. Finally, if that doesn't work, try `.ok_or(Report::msg)`.